src/Common/src/System/IO/PathInternal.Windows.StringBuffer.cs

```diff
@@ -15,27 +15,40 @@ internal static partial class PathInternal
         /// </summary>
         internal static bool IsExtended(StringBuffer path)
         {
-            Debug.Assert(path != null);
-            return path.StartsWith(ExtendedPathPrefix);
+            // While paths like "//?/C:/" will work, they're treated the same as "\\.\" paths.
+            // Skipping of normalization will *only* occur if back slashes ('\') are used.
+            return path.Length >= DevicePrefixLength
+                && path[0] == '\\'
+                && (path[1] == '\\' || path[1] == '?')
+                && path[2] == '?'
+                && path[3] == '\\';
         }
 
         /// <summary>
-        /// Returns true if the path uses the extended UNC syntax (\\?\UNC\)
+        /// Gets the length of the root of the path (drive, share, etc.).
         /// </summary>
-        internal static bool IsExtendedUnc(StringBuffer path)
+        internal unsafe static uint GetRootLength(StringBuffer path)
         {
-            Debug.Assert(path != null);
-            return path.StartsWith(UncExtendedPathPrefix);
+            if (path.Length == 0) return 0;
+            return GetRootLength(path.CharPointer, path.Length);
         }
 
         /// <summary>
-        /// Gets the length of the root of the path (drive, share, etc.).
+        /// Returns true if the path uses any of the DOS device path syntaxes. ("\\.\", "\\?\", or "\??\")
         /// </summary>
-        internal unsafe static uint GetRootLength(StringBuffer path)
+        internal static bool IsDevice(StringBuffer path)
         {
-            Debug.Assert(path != null);
-            if (path.Length == 0) return 0;
-            return GetRootLength(path.CharPointer, path.Length);
+            // If the path begins with any two separators is will be recognized and normalized and prepped with
+            // "\??\" for internal usage correctly. "\??\" is recognized and handled, "/??/" is not.
+            return IsExtended(path)
+                ||
+                (
+                    path.Length >= DevicePrefixLength
+                    && IsDirectorySeparator(path[0])
+                    && IsDirectorySeparator(path[1])
+                    && (path[2] == '.' || path[2] == '?')
+                    && IsDirectorySeparator(path[3])
+                );
```

Comment:
Yes, IsDirectorySeparator is either forward or backslash. Anything that starts with two separators will keep them and not add the current directory when they hit the the normalization routine, which is after the check for `\\?\` and `\??\`. As such `/\?\` will become `\\?\` and `/??/` will become something like `C:\??\`.